digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadwayChangesDAO.scala

```diff
@@ -380,28 +382,38 @@ class RoadwayChangesDAO {
             val news = ProjectDeltaCalculator.partition(delta.newRoads)
             news.foreach(roadwaySection => addToBatch(roadwaySection, AddressChangeType.New, roadwayChangePS, roadWayChangesLinkPS))
 
-            ProjectDeltaCalculator.partition(delta.unChanged.mapping).foreach { case (roadwaySection1, roadwaySection2) =>
-              addToBatchWithOldValues(roadwaySection1, roadwaySection2, AddressChangeType.Unchanged, roadwayChangePS, roadWayChangesLinkPS)
-            }
+            val unchanged = ProjectDeltaCalculator.partition(delta.unChanged.mapping)
+
+            val transferred = ProjectDeltaCalculator.partition(delta.transferred.mapping, terminated ++ news)
+
+            val numbering = ProjectDeltaCalculator.partition(delta.numbering.mapping)
 
-            ProjectDeltaCalculator.partition(delta.transferred.mapping, terminated ++ news).foreach { case (roadwaySection1, roadwaySection2) =>
+            val adjustedUnchanged = ProjectDeltaCalculator.adjustStartSourceAddressValues(unchanged._1, unchanged._2++transferred._2++numbering._2)
```

Comment:
In this code section we use too many times '._1' and '._2'. We can create a case class for those objects